src/components/NavigationFrame.js

```diff
@@ -177,16 +179,24 @@ function WalletSelector() {
           </MenuItem>
         ))}
         <Divider />
-        <MenuItem
+        {allowNewAccounts && <MenuItem
           onClick={() => {
             setAnchorEl(null);
-            setWalletIndex(addresses.length);
+            selectWallet(addresses.length);
           }}
         >
           <ListItemIcon className={classes.menuItemIcon}>
             <AddIcon fontSize="small" />
           </ListItemIcon>
           Create Account
+        </MenuItem>}
+
+        <MenuItem
+          onClick={() => {
```

Comment:
this should be 
```
onClick={() => {
  setAnchorEl(null);
  logout();
}}
```

null is un-necessary for the logout function, and setting the dropdown state to null when we logout fixes the dropdown rendering opened in a weird place when we log back in